googleapiclient/http.py

```diff
@@ -468,7 +469,7 @@ def __init__(self, body, mimetype='application/octet-stream',
     resumable: bool, True if this is a resumable upload. False means upload
       in a single request.
     """
-    fd = StringIO.StringIO(body)
+    fd = BytesIO(body)
```

Comment:
Both of `six.BytesIO` and `six.StringIO` are `cStringIO.StringIO` on Python 2.
I prefer `io` for more strict behavior. But current code is OK, too.